src/OrleansServiceBus/Providers/Streams/EventHub/EventHubPartitionCheckpointEntity.cs

```diff
@@ -0,0 +1,36 @@
+﻿
+using System;
+using Microsoft.ServiceBus.Messaging;
+using Microsoft.WindowsAzure.Storage.Table;
+
+namespace Orleans.ServiceBus.Providers
+{
+    internal class EventHubPartitionCheckpointEntity : TableEntity
+    {
+        public string Offset { get; set; }
+
+        public EventHubPartitionCheckpointEntity()
+        {
+            Offset = EventHubConsumerGroup.StartOfStream;
+        }
+
+        public static EventHubPartitionCheckpointEntity Create(string streamProviderName, string deploymentId, string partition)
+        {
+            return new EventHubPartitionCheckpointEntity
+            {
+                PartitionKey = MakePartitionKey(streamProviderName, deploymentId),
+                RowKey = MakeRowkey(partition)
+            };
+        }
+
+        public static string MakePartitionKey(string streamProviderName, string deploymentId)
+        {
+            return String.Format("EHCheckpoints_{0}_{1}", streamProviderName, deploymentId);
+        }
+
+        public static string MakeRowkey(string partition)
+        {
+            return String.Format("partition_{0}", partition);
```

Comment:
Readability of rowkey name.  partition_112 is more clear than 112.
I'm not attached to this, so if you'd prefer we save space, I'll change it.